base_layer/core/src/transactions/key_manager/inner.rs

```diff
@@ -532,28 +531,13 @@ where TBackend: KeyManagerBackend<PublicKey> + 'static
     }
 
     async fn get_private_comms_key(&self) -> Result<PrivateKey, KeyManagerServiceError> {
-        let branch = TransactionKeyManagerBranch::Spend.get_branch_key();
+        let branch = TransactionKeyManagerBranch::Comms.get_branch_key();
         let index = 0;
-
-        match self.wallet_type {
-            WalletType::DerivedKeys | WalletType::ProvidedKeys(_) => {
-                self.get_private_key(&TariKeyId::Managed {
-                    branch: branch.clone(),
-                    index,
-                })
-                .await
-            },
-            WalletType::Ledger(_) => {
-                let km = self
-                    .key_managers
-                    .get(&branch)
-                    .ok_or_else(|| self.unknown_key_branch_error("get_private_comms_key", &branch))?
-                    .read()
-                    .await;
-                let key = km.get_private_key(index)?;
-                Ok(key)
-            },
-        }
+        self.get_private_key(&TariKeyId::Managed {
+            branch: branch.clone(),
+            index,
+        })
```

Comment:
Something doesn't sit right about this.

Just knowing how we flipped on this implementation before. It just doesn't *seem* correct and I can't put my finger on it.